libevm/VM.cpp

```diff
@@ -91,10 +91,36 @@ VM::VM()
         aleth_get_buildinfo()->project_version,
         ::destroy,
         ::execute,
+        setTracer,
         nullptr,
     }
 {}
 
+void VM::setTracer(
+    evmc_instance* _instance, evmc_trace_callback _callback, evmc_tracer_context* _context) noexcept
+{
+    auto vm = static_cast<dev::eth::VM*>(_instance);
+    vm->m_trace = _callback;
+    vm->m_traceContext = _context;
+}
+
+void VM::trace() noexcept
+{
+    if (m_trace)
+    {
+        auto const& metrics = c_metrics[static_cast<size_t>(m_OP)];
+        evmc_uint256be topStackItem;
+        evmc_uint256be const* pushedStackItem = nullptr;
+        if (metrics.num_stack_returned_items == 1)
```

Comment:
Should this have an assert for >1? Should it just pick the top item in case of >1? (I don't even think there is a single instruction currently which would do that though.)

Reply:
`DUPN` / `SWAPN` have `N` returned items there (not sure why `DUPN` do)

Maybe returning the top item in case `>1` is a good idea.

Reply:
Yes, that's good idea.